pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -268,13 +268,18 @@ cdef class _Timestamp(datetime):
                                  "without freq.")
             return Timestamp((self.freq * other).apply(self), freq=self.freq)
 
-        elif PyDelta_Check(other) or hasattr(other, 'delta'):
-            # delta --> offsets.Tick
+        elif PyDelta_Check(other):
             nanos = delta_to_nanoseconds(other)
             result = Timestamp(self.value + nanos,
                                tz=self.tzinfo, freq=self.freq)
+            return result
+
+        elif hasattr(other, 'delta'):
```

Comment:
You might be able to import `tslibs.offsets._Tick` here and make this an isinstance check (thought that might cause a circular import, not sure off the top)